packages/aws_common/lib/src/io/aws_file_platform_io.dart

```diff
@@ -104,4 +108,34 @@ class AWSFilePlatform extends AWSFile {
     // the constructors ensures it won't reach to this point, but just in case
     throw const InvalidFileException();
   }
+
+  @override
+  Future<String?> get contentType async {
```

Comment:
Should be wrapped in an `AsyncMemoizer` or made `late final` since this should only run once.

Reply:
Same for web